Compressed cleanup

huss opened the pull request:
# Description

This is a maintenance PR that cleans up the code but does not change any functionality. It should only be merged after PR #740 as it continues that work. At this time all the changes in that PR are also listed in this one. The main edits are:

- Old code & functions before compressed readings (fast-pt) that were no longer used are removed.
- Code, functions & views associated with compressed readings for lines were removed as they are now all replaced.
- All file names and references to compressed were removed when they could be. Given the changes in the code it now names/refers to averaged, readings, etc.
- Migrations were added for the removed items from the DB.
- I previously used line before meter/group in names for the new unit related functions. This mimicked the order in the state. However, there already is a lot of code with it the other way around so I switched my recent code.

## Type of change

- [ ] Note merging this changes the node modules
- [x] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

- I had OED build the DB from scratch and it worked. However, the migrations are not yet tested.
- There is a merge conflict. Once PR #740 is merged, this will be fixed if still an issue.

Most of these should have a TODO in the code.

- Bar still has compressed in it as the needed changes for unit is not yet done.
- The index for the materialized view does not include hourly and it needs to checked out.
- Test code was not extensively edited and still needs to be done for new code.
- The baseline code is out of date. It will need to be extensively edited when baselines are incorporated into OED. As such, it was not edited at this time.

huss committed:
start removal of old readings code

- remove old compress reading functions from DB
- remove/drop old views in DB
- rename refresh functions without compressed and with time length
- remove src/server/sql/reading/generateCompressedReadingViews.js since
no longer needed as for old compressed readings
- update test for functions used but not what test so may not work

huss committed:
remove unused line/bar reading code & fix names

- Removed unused types/routes for Line/BarReadings and related routes
- Code had CompareReading instead of CompareReadings as done with other
names so changed

huss committed:
remove unused line/bar reading functions & compressed from file names

- Old SQL/functions related to old way to doing readings was removed.
A new migration for dropping the removed functions was added. Another
migration to drop old views was added.
- remove compressed from all file names except test (coming)
and migrations. Moved what was in compressed-readings.ts
into existing readings.ts.
- Noted test files that probably need to rename when tests updated.
- small fixes

huss committed:
remove compressed word from code

This is the first pass at removing the word compressed from the code,
except for these places
- Old conversions where needed.
- Code still needing update for units for bar.
- Test code that will be done later.
- Baseline that needs a major upgrade before actually used.

huss committed:
use meter/group then line

In my PR to incorporate units, I chose to use meter/group first and then
line in names since that followed the order in the state. However, a lot
of existing code does it the other way around so this inverts my naming
to be consistent with that. It also uses singular as in the rest of the
code.

truongdd03 commented on src/server/models/Reading.js:
```diff
@@ -245,158 +207,57 @@ class Reading {
-	 * @param conn is the connection to use.
-	 * @return {Promise<object<int, array<{reading_sum: number, start_timestamp: Date, end_timestamp: Date}>>>}
-	 */
-	static async getBarchartReadings(meterIDs, duration, fromTimestamp = null, toTimestamp = null, conn) {
```
Could you please explain why this function is removed? I saw that this function is still used in src/server/test/db/barchartReadingsTests.js. Thank you.

huss replied:
Yes, you are correct. To move this forward faster, I have not been updating the test code (as noted in the PR). As a result, the test code does not reflect the new functions nor ones that were obsolete. This is going to happen later (hopefully not too long now). In this case, the test code either needs to have the function name change (that also means modifying the parameter list to deal with the new unit argument and testing it) if there is a new equivalent or removing the tests for obsolete code. Thus, it will take some effort to do all the test changes including creating the new tests for new capabilities. So, I felt it was best to clean up the code of obsolete code whose only use was by test code (which indicates it not really used by the project any more but I also did other checks to try to be careful). Doing OED code this way is unusual but acceptable right now since it isn't on the development branch (that should always pass the tests). The plan was this would allow for more rapid integration and use by the other teams working on resource generalization so faster progress in seeing results and finding any fundamental issues in this complex change. It does mean we are acquiring technical debt by not testing and some code may need future fixes. The plan is to pay down that debt soon and before we fully accept the new code as part of OED (as in merging into development). I hope this helps. I agree the motivation was not clear and it is unusual for OED. The design document for this tried to indicate that plan by making the full testing of the code independent of the implementation and somewhat later in the figure.

truongdd03 replied:
Thank you for your explanation. This makes sense to me.

truongdd03 commented on src/server/migrations/0.8.0-1.0.0/index.js:
```diff
@@ -10,6 +10,6 @@ module.exports = {
 	up: async db => {
 		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/create_cik_table.sql'));
-		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/create_compressed_reading_views.sql'));
+		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/create_reading_views.sql'));
```
I think we also need to call `drop_old_functions.sql` and `drop_old_views.sql` here. I don't have good knowledge of creating migrations so please correct me if I'm wrong. Thank you.

huss replied:
You are correct. I also did not have the subdirectories I added so I changed that too. When migrations are tested for 1.0 the final check will be made to be sure they all work.

truongdd03 commented on src/server/routes/readings.js:
```diff
@@ -32,53 +32,6 @@ function formatBarReadings(rows) {
 	return rows.map(row => [row.start_timestamp.valueOf(), row.reading_sum]);
```
I think the function `formatBarReadings` and `formatLineReadings` are no longer necessary. Do you think we should remove them? I can't make comments above this line so I put it here.

huss replied:
I agree. The new use is in unitReadings.js where formatLineReadings is now formatReadingRow but formatBarReading is the same name.

truongdd03 commented on src/server/migrations/0.8.0-1.0.0/sql/readings/drop_old_views.sql:
```diff
@@ -0,0 +1,6 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+DROP VIEW IF EXISTS daily_readings;
+DROP VIEW IF EXISTS hourly_readings;
```
I test this command manually and get the following error:
`ERROR:  "daily_readings" is not a view
HINT:  Use DROP MATERIALIZED VIEW to remove a materialized view.`
Do you think we should change this into `DROP MATERIALIZED VIEW IF EXISTS daily_readings;`? The same for hourly_readings.

huss replied:
Again, you are right. I looked at my notes in the resourceGeneralization devDoc and it has MATERIALIZED. I think I put this in, found the issue while manually working on the DB and then forgot to change this. Good catch.

huss committed:
fix issues noted in review

- migrations not properly listed in index and how drop view
- remove unneed functions in reading.js

truongdd03 commented on src/server/migrations/0.8.0-1.0.0/index.js:
```diff
@@ -9,7 +9,9 @@ module.exports = {
 	up: async db => {
-		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/create_cik_table.sql'));
-		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/create_compressed_reading_views.sql'));
+		await db.none(sqlFile('../migrations/0.8.0-1.0.0/sql/cik/create_cik_table.sql'));
```
I think the repository `src/server/migrations/0.8.0-1.0.0/sql/cij` has been accidentally mislabeled. We should change it into `cik` for this line to work.

huss replied:
Yes. I have renamed the directory to cik.

huss committed:
fix migration directory name

truongdd03 reviewed (approved):
@huss Everything looks good. This PR is ready to merge.

huss committed:
Merge branch 'resourceGeneralization' of https://github.com/OpenEnergyDashboard/OED into pr/huss/741

huss committed:
fix ESLint issue

Changed Promise{} to Primise<unknown> per ESLint suggestion but note
that other seems to use void. Testing found no issue.

huss commented:
I merged in resourceGeneralization and found no issues. Tests are not yet fixed so they will fail. This should be ready to merge. I note that @truongdd03 reviewed and approved this PR.
